src/Microsoft.DotNet.Darc/src/DarcLib/Models/Darc/DependencyDetail.cs

```diff
@@ -6,18 +6,69 @@ namespace Microsoft.DotNet.DarcLib
 {
     public class DependencyDetail
     {
-        public string Branch { get; set; }
-
+        public DependencyDetail() { }
+        public DependencyDetail(DependencyDetail other)
+        {
+            Name = other.Name;
+            Version = other.Version;
+            RepoUri = other.RepoUri;
+            Commit = other.Commit;
+            Pinned = other.Pinned;
+            Type = other.Type;
+            CoherentParentDependencyName = other.CoherentParentDependencyName;
+        }
+        /// <summary>
+        ///     Name of dependency.
+        /// </summary>
         public string Name { get; set; }
 
+        /// <summary>
+        ///     Version of dependency.
+        /// </summary>
         public string Version { get; set; }
 
+        /// <summary>
+        ///     Source repository uri that the dependency was produced from.
+        /// </summary>
         public string RepoUri { get; set; }
 
+        /// <summary>
+        ///     Source commit that the dependency was produced from.
+        /// </summary>
         public string Commit { get; set; }
 
+        /// <summary>
+        ///     True if the dependency should not be updated, false otherwise.
+        /// </summary>
         public bool Pinned { get; set; }
 
+        /// <summary>
+        ///     Type of dependency (e.g. Product or Toolset).
+        /// </summary>
         public DependencyType Type { get; set; }
+
+        /// <summary>
+        ///     Another dependency for which this dependency must be coherent with.
+        ///     This means:
+        ///     If I have 3 repositories which have a potentially incoherent dependency structure:
+        ///     A
+        ///     |\
+        ///     B |
+        ///     \ |
+        ///      C
+        ///     A different version of C could appear in A and B.
+        ///     This may not be a problem, or it could be undesirable.
+        ///     This can be resolved to be always coherent by identifying that A's dependency on C
+        ///     must be coherent with parent B. Specifically, this means that the build that produced B must
+        ///     also have an input build that produced C.
+        ///     
+        ///     Concretely for .NET Core, core-setup has a dependency on Microsoft.Private.CoreFx.NETCoreApp produced
+        ///     in corefx, and Microsoft.NETCore.Runtime.CoreCLR produced in coreclr.  corefx has a dependency on
+        ///     Microsoft.NETCore.Runtime.CoreCLR. This means that when updating Microsoft.Private.CoreFx.NETCoreApp
+        ///     in core-setup, also update Microsoft.NETCore.Runtime.CoreCLR to the version used to produce that
+        ///     Microsoft.Private.CoreFx.NETCoreApp. By corrolary, that means Microsoft.NETCore.Runtime.CoreCLR cannot
+        ///     be updated unless that version exists in the subtree of Microsoft.Private.CoreFx.NETCoreApp.
+        /// </summary>
+        public string CoherentParentDependencyName { get; set; }
```

Comment:
>CoherentParentDependencyName [](start = 22, length = 28)

What is the value of this... is it "Microsoft.NETCore.Runtime.CoreCLR" or "corefx", or something else? I don't know if a "Dependency" is an asset, a build, a branch, or a repository (or some other thing). It should probably use one of those four words if it's one of those 4 things.

Reply:
Name of a dependency.  The same as the names of the <Dependency> nodes listed in the file.

```
<Dependency Name="Microsoft.DotNet.Arcade.Sdk" Version="1.0.0-beta.19151.1">
      <Uri>https://github.com/dotnet/arcade</Uri>
      <Sha>0951c0fd4c38a6d01affe53b042111f8f7b91e28</Sha>
    </Dependency>
```

Reply:
In other words, we use 'Dependency' in this file, and this attribute should point to another Dependency.